### PR TITLE
Review and update Standard Types 

### PR DESCRIPTION
**Pull request by tpdorsey**
Review standard types documentation.

## Change Summary

- Update and add links
- Move Constrained types
- Remove duplicated Strict types
- Minor edits

## Related issue number

Closes PYD-78

## Checklist

* [ ] Unit tests for the changes exist
* [ ] Tests pass on CI and coverage remains at 100%
* [x] Documentation reflects the changes where applicable
* [ ] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)
* [x] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

Selected Reviewer: @lig

### CODE REVIEW

**Comment by tpdorsey** [comment]
please review

File: docs/usage/types/strict_types.md [hunk 1]
```diff
@@ -57,3 +62,89 @@ m = Model(finite=1.0)
 print(m)
 #> finite=1.0
 ```
+
+### Arguments to `conbytes`
```
Comment (Kludex) [inline docs/usage/types/strict_types.md:66]: I don't think it's a good idea to have these "Arguments" sections. It's pretty much the same as sharing the API documentation.

**Review by Kludex — CHANGES_REQUESTED** [review]

File: docs/usage/types/strict_types.md (hunk 1 above)
Comment (Kludex) [reply docs/usage/types/strict_types.md:66]: They can easily become out-dated.

Comment (lig) [reply docs/usage/types/strict_types.md:66]: Just a link to the API docs then? Do API docs have the same info on them?

Comment (Kludex) [reply docs/usage/types/strict_types.md:66]: Yes, and yes (if not, they should).

Comment (tpdorsey) [reply docs/usage/types/strict_types.md:66]: Yes the api docs have this, I'll revert to what we had previously

**Comment by tpdorsey** [comment]
please review

**Review by lig — APPROVED** [review]
👍🏻

File: docs/usage/types/strict_types.md [hunk 2]
```diff
@@ -1,5 +1,8 @@
 Strict types enable you to prevent coercion from compatible types.
 
+## Strict types
+
+
 Pydantic provides the following strict types:
 
 - [`StrictBool`][pydantic.types.StrictBool]
```
Comment (adriangb) [inline docs/usage/types/strict_types.md:8]: I feel like we should deprecate these. It seems much more composable to recommend`Strict[bool]`.

Comment (tpdorsey) [reply docs/usage/types/strict_types.md:8]: @dmontagu thoughts?

Comment (adriangb) [reply docs/usage/types/strict_types.md:8]: I guess I didn't look but I expected that to just work.

Comment (lig) [reply docs/usage/types/strict_types.md:8]: I'd suggest extracting this idea into a new issue. Looks like a legitimate improvement that could be implemented in Pydantic V2.1+

Comment (tpdorsey) [reply docs/usage/types/strict_types.md:8]: PYD-156

**Review by adriangb — APPROVED** [review]

**Review by Kludex — APPROVED** [review]

**Pull request merged**